kroxylicious-filters/kroxylicious-record-encryption/pom.xml

```diff
@@ -144,6 +147,29 @@
 
     <build>
         <plugins>
+            <plugin>
+                <groupId>io.kroxylicious</groupId>
+                <artifactId>kroxylicious-schema-maven-plugin</artifactId>
+                <version>${project.version}</version>
+                <executions>
+                    <execution>
+                        <id>generate-config-model</id>
+                        <goals>
+                            <goal>compile-schema</goal>
+                        </goals>
+                        <phase>generate-sources</phase>
+                        <configuration>
+                            <failOnWarnings>true</failOnWarnings>
+                            <headerFile>../../etc/license.txt</headerFile>
+                            <existingClasses>
+                                <io.kroxylicious.filter.encryption.config.RecordEncryptionConfigKmsConfig>java.lang.Object</io.kroxylicious.filter.encryption.config.RecordEncryptionConfigKmsConfig>
+                                <io.kroxylicious.filter.encryption.config.RecordEncryptionConfigSelectorConfig>java.lang.Object</io.kroxylicious.filter.encryption.config.RecordEncryptionConfigSelectorConfig>
+                                <io.kroxylicious.filter.encryption.config.RecordEncryptionConfigExperimental>java.util.Map&lt;String, Object&gt;</io.kroxylicious.filter.encryption.config.RecordEncryptionConfigExperimental>
```

Comment:
I ended up keeping the `experimental` property as a `Map<String, Object>`, but at one point I had it as a generated class. 

I don't recall now exactly our reasons for it being a map, @robobario do you? Is now a good time to make some of them not experimental?

Reply:
https://github.com/kroxylicious/kroxylicious/pull/1032 is where it arrived as a Map

All I can dig up is this convo:
```
tbentley: I think we need a better mechanism for exposing config without committing to it as a supported interface.
tbentley: A bit like HTTP's X headers
tbentley: Jackson supports mapping additional properties to a Map
tbentley: Maybe we should use that?
Sam B: isn’t the problem more that once the structure is written in a yaml file people expect it to keep working
Sam B: So we would maybe need a custom or overrides node and specify that is a grab bag of things?
Sam B: as your right we do need more flexibility here
tbentley: Yeah, that's the sort of thing... have an experimentalFoo property, or unsupportedProperties or something.
tbentley: dontUseThisExceptForDemos
Sam B: theEscapeHatch
Sam B: hereBeDragons
```

We don't have to implement it as a `Map`, it could be an object that declares those fields and allows additional properties.

> Though I don't really remember why we thought they were experimental and thus not part of the public API commitment.

The negative caching for alias resolution is/(was?) a stopgap, I think it was just a construction site and we didn't want to commit to these with the same level of guarantee as in our [deprecation policy](https://github.com/kroxylicious/kroxylicious/blob/main/DEV_GUIDE.md#deprecation-policy), as we may have rapidly removed these configs. However they're still here, some should be elevated to the schema'd area.

 I think in discussion we decided that the `experimental` naming was enough to convey it didn't obey the deprecation policy, maybe we could add something explicit to the docs though.

I think `maxRecordsForEncryptions` should be moved into the schema'd classes. Probably some cache configurations too.

Reply:
if we can collect all unknown properties into a Map that would be cool too, you could log once saying "here's all the unknown stuff you configured, check the docs for removals of experimental configs"